nautobot/extras/utils.py

```diff
@@ -87,3 +88,23 @@ def wrapper(model_class):
         return model_class
 
     return wrapper
+
+
+class TemplateRenderer:
```

Comment:
This is cool but I feel like it should live somewhere else since you're saying it is "to be used anywhere that Nautobot renders Jinja2 templates". Perhaps putting it adjacent to `nautobot.utilities.utiles.render_jinja2()` or replacing that function or making the `render_jinja2()` function use this?